### PR TITLE
Added LayoutChangedNotifier.

### PR DESCRIPTION
**Pull request by dragostis**
Added a simple widget that automatically dispatches a
LayoutChangedNotification when its child changes layout.

### CODE REVIEW

File: packages/flutter/lib/src/material/material.dart [hunk 1]
```diff
@@ -656,5 +656,43 @@ class _InkHighlight extends InkFeature implements InkHighlight {
...
     }
   }
+}
+
+/// A widget that automatically dispatches a [LayoutChangedNotification] when
+/// the layout of its child changes.
+///
+/// Useful especially when having some complex, layout-changing animation within
+/// [Material] that is also interactive.
+class LayoutChangedNotifier extends SingleChildRenderObjectWidget {
```
Comment (abarth) [inline packages/flutter/lib/src/material/material.dart]: I'd put this in its own file.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/material.dart [hunk 2]
```diff
@@ -656,5 +656,43 @@ class _InkHighlight extends InkFeature implements InkHighlight {
...
+  _RenderLayoutChangedWithCallback({
+    RenderBox child,
+    this.onLayoutChangedCallback
+  }) : super(child);
 
+  VoidCallback onLayoutChangedCallback;
+
+  @override
+  void performLayout() {
+    if (onLayoutChangedCallback != null)
```
Comment (Hixie) [inline packages/flutter/lib/src/material/material.dart]: This doesn't work. There's no way to guarantee that you'll be called whenever the child changes layout. For example, this won't work if the child is just transforming the child (RenderTransform), or if the children are being moved without the constraints changing, etc.

**Comment by Hixie** [comment]
I don't think we should check this in. This doesn't catch all layout changes, it only catches size changes. In general there's no way to guarantee that you'll be called whenever the child changes layout. For example, this won't work if the child is just transforming the child (RenderTransform), or if the children are being moved without the constraints changing, etc.

**Comment by Hixie** [comment]
We discussed this; it might still be useful as specifically a size-change notifier, if it had a name that matched that specific behavior.

The RenderObject should be at the rendering/ layer and the widget at the widgets/ layer.

**Comment by Hixie** [comment]
fix docs, make the notification be a subclass, and LGTM

**Pull request merged**